### DISCUSSION TITLE
Transmitting data using Radiomaster PR1

### DISCUSSION THREAD

**Discussion by DawnDawn98** [ExpressLRS/ExpressLRS#2854] [Q&A]
Hello,
I am setting up a link between a Radiomaster boxer unit and a PR1 unit connected to my laptop through a COM port. I was able to bind the two, and when I access the serial port it seems that I get data bytes from the port, but when I am trying to send data (I am just sending FF to the com port repeatedly), it doesn't seem to be transmitting it. I don't get any indication that the data was transmitted..  Am I missing some protocol requirement when sending data to the PR1 unit over the UART connection of the COM port?

**Comment by pkendall64** [comment]
Yes you have to talk CRSF protocol with the receiver.
See https://github.com/crsf-wg/crsf/wiki

**Reply by DawnDawn98** [reply]
Thank you for the answer. If I understand correctly, the CRSF protocol is relevant for the communication between the MCU and the transceiver, which is done inside the PR1 chipset. I am trying to communicate with the MCU from "the outside". I am "pretending" to be the flight controller basically, I want to send the MCU data which it will then "pack" according to the CRSF protocol and send to the transmitter, which will transmit it to the Boxer unit. Does this make sense?

**Reply by pkendall64** [reply]
No the CRSF communication is between the MCU and the flight controller. Unless of course you configure the receiver to talk SBUS, but then you have no telemetry. The protocol between the MCU and the transceiver is the ELRS OTA protocol.
